darwin/Classes/JustWaveformPlugin.m

```diff
@@ -121,7 +121,7 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
             UInt32 scaledSampleIdx = 0;
             int progress = 0;
 
-            while (frameCount > 0) {
+            while (frameCount > 0 && progress < 100) {
```

Comment:
Just curious, what is the scenario where the second condition is required?

Reply:
Sometimes the progress never become 100, kind of rare times. may be frame count cause the problem. Its difficult to find

Reply:
Your condition wouldn't help if the progress never became 100. Maybe you meant that frameCount never becomes 0?